### ISSUE TITLE
Rule Change: prefer-const - new option to ignore arrays and objects

### ISSUE THREAD

**Issue by saturdaywalkers** [eslint/eslint#16547]
### What rule do you want to change?

prefer-const

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
// new option prefer-const scalar-only: true
let a = 1; // error, should be const

let b = { aa:1, bbb:2 }; // ok, as...
b.aa = 'fred'; // legal, b is not really a constant

let c = [ 1,2,3]; // ok, as...
c.push( 4 ); // legal, so c is not really a constant
```

### What does the rule currently do for this code?

does not discriminate between const var = scalar, array, or object

### What will the rule do after it's changed?

allow prefer-const to skip arrays, and objects

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by amareshsm** [comment]
Giving the option to skip the array and object doesn't make sense.

```
let b = { aa:1, bbb:2 }; // ok, as...
b.aa = 'fred'; // legal, b is not really a constant

let c = [ 1,2,3]; // ok, as...
c.push( 4 ); // legal, so c is not really a constant
```

In the above code, technically b and c are constants. In the case of objects/arrays, constants can't be reassigned/redeclared it can be updated. Pls, refer to the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const).

**Comment by saturdaywalkers** [comment]
@amareshsm To me, while legal js, `const b={}; b.aa = 1` seems wrong. I'd like to enforce `prefer_const` for scalars, but not for objects/arrays

**Comment by mdjermanovic** [comment]
I'm not in favor of adding this option because it's not in line with the semantics of `const` in JavaScript and is therefore a stylistic preference. You can always copy the core `prefer-const` rule into a custom rule and make it behave the way you want.

**Comment by amareshsm** [comment]
you can use [`eslint-plugin-functional`](https://github.com/eslint-functional/eslint-plugin-functional) plugin to disable mutations.
